src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -107,6 +108,7 @@ public OpenDistroSecurityFilter(final PrivilegesEvaluator evalp, final AdminDNs
         this.cs = cs;
         this.compatConfig = compatConfig;
         this.indexResolverReplacer = indexResolverReplacer;
+        this.immutableIndicesMatcher = immutableIndicesMatcher;
```

Comment:
nit: construct `WildcardMatcher` inside `OpenDistroSecurityFilter` to fully encapsulate implementation details (whether it is implemented as `WildcardMatcher` or differently is internal to `OpenDistroSecurityFilter`). Move logging of `immutableIndices` to the `OpenDistroSecurityFilter` as well.